foa_mesh/src/runner/management.rs

```diff
@@ -0,0 +1,383 @@
+            authentication_protocol_identifier: ieee80211::elements::mesh::MeshConfigurationAuthenticationProtocolIdentifier::NoAuthentication,
+            mesh_formation_info: ieee80211::elements::mesh::MeshFormationInfo::new()
+                .with_connected_to_mesh_gate(false) // TODO fill this in once we have it
+                .with_num_peerings(0) // TODO fill this in
+                .with_connected_to_as(false), // 'Connected to authentication system' is always false in open / SAE mesh
+            mesh_capability: ieee80211::elements::mesh::MeshCapability::new()
+                .with_accept_additional_mesh_peerings(true) // TODO fill this in
+                .with_forwarding(true)
+        }
+    }
+
+    fn does_mesh_sta_configuration_match(elements: ReadElements) -> bool {
+        // Check if mesh profile is equal
+        if (elements
+            .get_first_element::<MeshIDElement>()
+            .map(|a| (a.ssid() != MESH_ID)))
+        .unwrap_or(true)
+        {
+            return false;
+        }
+
+        let Some(peer_config_element) = elements.get_first_element::<MeshConfigurationElement>()
+        else {
+            debug!("no mesh configuration element");
+            return false;
+        };
+        let own_config_element = Self::generate_own_mesh_configuration_element();
+        if peer_config_element.active_path_selection_metric_identifier
+            != own_config_element.active_path_selection_metric_identifier
+            || peer_config_element.active_path_selection_protocol_identifier
+                != own_config_element.active_path_selection_protocol_identifier
+            || peer_config_element.authentication_protocol_identifier
+                != own_config_element.authentication_protocol_identifier
+            || peer_config_element.congestion_control_mode_identifier
+                != own_config_element.congestion_control_mode_identifier
+            || peer_config_element.syncronization_method_identifier
+                != own_config_element.syncronization_method_identifier
+        {
+            debug!("mesh configuration element mismatch");
+            return false;
+        }
+        // TODO we should also check the peers EPD capability; but this is very uncommon on 2.4GHz or 5GHz
+
+        // Check if all other fields of the mesh STA configuration matches
+        if (elements
+            .get_first_element::<SupportedRatesElement>()
+            .map(|a| (a != DEFAULT_SUPPORTED_RATES)))
+        .unwrap_or(true)
+        {
+            debug!("default rates mismatch");
+            return false;
+        }
+        if elements
+            .get_first_element::<ExtendedSupportedRatesElement>()
+            .map(|a| (a != DEFAULT_XRATES))
+            .unwrap_or(true)
+        {
+            debug!("extended rates mismatch");
+            return false;
+        }
```

Comment:
I'm not sure, if it makes sense to check if the entire element matches byte for byte, since IIRC we technically only need to know the highest rate supported by a peer and a complete match is pretty unlikely.

Reply:
I do think we need to check byte for byte, as specified in 14.2.4 Mesh STA configuration in the standard:

> The BSSBasicRateSet parameter of the MLME-START.request primitive is identical to the basic
> rate set indicated by the Supported Rates and BSS Membership Selectors element and Extended
> Supported Rates and BSS Membership Selectors element, if present, received in the MLME-
> MESHPEERINGMANAGEMENT.indication primitive.

Reply:
Ok, interesting. This does appear to be in spec. I'm wondering why though. Wouldn't it make sense for some mesh STAs to have different data rates then others?

Reply:
Seems weird, since that would cause mesh STAs to either converge to the greatest common denominator or exclude some STAs.